bokehjs/test/integration/regressions.ts

```diff
@@ -3775,4 +3775,26 @@ describe("Bug", () => {
       await view.ready
     })
   })
+
+  describe("in issue #13806", () => {
+    it("doesn't allow to change the order of LayoutDOM.children", async () => {
+      const child0 = new Spacer({width: 50, height: 50, styles: {background_color: "red"}})
+      const child1 = new Spacer({width: 50, height: 50, styles: {background_color: "green"}})
+      const child2 = new Spacer({width: 50, height: 50, styles: {background_color: "blue"}})
+      const layout = new Row({children: [child0, child1, child2]})
+      const {view} = await display(layout, [200, 100])
+      layout.children = [child2, child0, child1]
+      await view.ready
+    })
+
+    it("doesn't allow to change the order of Pane.elements", async () => {
+      const child0 = new Pane({styles: {width: "50px", height: "50px", background_color: "red"}})
+      const child1 = new Pane({styles: {width: "50px", height: "50px", background_color: "green"}})
+      const child2 = new Pane({styles: {width: "50px", height: "50px", background_color: "blue"}})
+      const layout = new Pane({elements: [child0, child1, child2], styles: {display: "flex", flex_direction: "row"}})
+      const {view} = await display(layout, [200, 100])
+      layout.elements = [child2, child0, child1]
```

Comment:
nit: if the names were `red`, `green`, `blue` instead of `child0`, `child1`, `child2` then it would be trivial to compare the test image to the code at a glance